internal/storage/v2/memory/memory.go

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package memory
+
+import (
+	"context"
+	"sync"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	conventions "go.opentelemetry.io/collector/semconv/v1.16.0"
+
+	v1 "github.com/jaegertracing/jaeger/internal/storage/v1/memory"
+	v2api "github.com/jaegertracing/jaeger/internal/storage/v2/api/tracestore"
+	"github.com/jaegertracing/jaeger/internal/tenancy"
+)
+
+const noServiceName = "OTLPResourceNoServiceName"
+
+// Store is an in-memory store of traces
+type Store struct {
+	sync.RWMutex
+	// Each tenant gets a copy of default config.
+	// In the future this can be extended to contain per-tenant configuration.
+	defaultConfig v1.Configuration
+	perTenant     map[string]*Tenant
+}
+
+// Tenant is an in-memory store of traces for a single tenant
+type Tenant struct {
```

Comment:
I would prefer to move Tenant into `tenant.go`, the code is messy when you have two structs mixed up